### PR TITLE
Add path __eq__ and __repr__ QOL changes

### PR DESCRIPTION
**Pull request by Schamper**
Allow comparison to strings, as well as remove the class name from the repr output.

### CODE REVIEW

File: flow/record/fieldtypes/__init__.py [hunk 1]
```diff
@@ -609,6 +609,14 @@ def __new__(cls, *args):
 
         return cls._from_parts(args)
 
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, str):
+            return str(self) == other
```
Comment (yunzheng) [inline flow/record/fieldtypes/__init__.py:614]: I would also expect the following unit test to pass:

```python
from flow.record.fieldtypes import windows_path

def test_windows_path_eq():
    path = windows_path(r"c:\windows\test.exe")
    assert path == r"c:\windows\test.exe"
    assert path == r"c:/windows/test.exe"
    assert path == r"c:/windows\test.exe"
```

I think we could do fast pass str compare and otherwise initialize path (which is more expensive):
```suggestion
            return str(self) == other or self.__class__(other)
```

**Review by yunzheng — CHANGES_REQUESTED** [review]
Also, since we are changing the repr output, would we prefer:

repr output:
`<test/path path="c:/windows/test.exe">`

or:

str output:
`<test/path path="c:\windows\test.exe">`

**Comment by Schamper** [comment]
> Also, since we are changing the repr output, would we prefer:
> 
> repr output: `<test/path path="c:/windows/test.exe">`
> 
> or:
> 
> str output: `<test/path path="c:\windows\test.exe">`

I thought about this before and I think I prefer the latter (str output). However to make the quoting of `repr()` work correctly I had to emulate that behaviour in pure Python. Using `repr()` directory would result in the `\\` getting escaped to `\\\\`.

File: flow/record/fieldtypes/__init__.py [hunk 2]
```diff
@@ -640,4 +648,13 @@ class posix_path(pathlib.PurePosixPath, path):
 
 
 class windows_path(pathlib.PureWindowsPath, path):
-    pass
+    def __repr__(self) -> str:
+        s = str(self)
+        quote = "'"
+        if "'" in s:
+            if '"' in s:
+                s = s.replace("'", "\\'")
```
Comment (yunzheng) [inline flow/record/fieldtypes/__init__.py:656]: I took the liberty to add some quotation tests, but i missed this one. I think this case happens if there are both single and double quotes in the path.

If you can add a test for this one it should be good to go

**Review by yunzheng — CHANGES_REQUESTED** [review]

**Review by yunzheng — APPROVED** [review]

**Comment by yunzheng** [comment]
LGTM

**Review by pyrco — CHANGES_REQUESTED** [review]

File: flow/record/fieldtypes/__init__.py [hunk 3]
```diff
@@ -640,4 +648,13 @@ class posix_path(pathlib.PurePosixPath, path):
...
+            else:
+                quote = '"'
+
+        return f"{quote}{s}{quote}"
```
[the remark this answers is not in the document]
Comment (Schamper) [reply flow/record/fieldtypes/__init__.py:660]: It shouldn't? It uses the `__repr__` defined in the base `path` class here.

[the remark this answers is not in the document]
Comment (yunzheng) [reply flow/record/fieldtypes/__init__.py:660]: the quotation handling only happens for windows_path though, it should also apply for posix_path or path in general?  There are also tests missing for posix_path quotation edge cases.

[the remark this answers is not in the document]
Comment (yunzheng) [reply flow/record/fieldtypes/__init__.py:660]: hmm then I probably don't understand why you have to emulate repr for windows_path but not for posix_path. I thought it was to get a consistent single or doublequoted string.

[the remark this answers is not in the document]
Comment (Schamper) [reply flow/record/fieldtypes/__init__.py:660]: Because for Windows paths it will also escape the "\" to "\\" (or "\\\\" if you want to go by the correct notation, which will print as "\\"). IMO this is not wanted (you want to print a single backslash), so I didn't want to use repr() directly, hence the emulation of how repr() handles quotes in strings internally.

[the remark this answers is not in the document]
Comment (yunzheng) [reply flow/record/fieldtypes/__init__.py:660]: gotcha, thanks for explaining!

[the remark this answers is not in the document]
Comment (pyrco) [reply flow/record/fieldtypes/__init__.py:660]: > For posix I just call `repr(str(self))` currently, so Python handles that for us with the call to `repr()`.

AH I missed the base `path` class changes to `__repr__` apparently.

LGTM

**Review by pyrco — APPROVED** [review]
